kubemarine/procedures/upgrade.py

```diff
@@ -137,6 +138,22 @@ def upgrade_containerd(cluster: KubernetesCluster):
                 if isinstance(value, dict):
                     config_string += f"\n[{key}]\n{toml.dumps(value)}"
             utils.dump_file(cluster, config_string, 'containerd-config.toml')
+            
+            node_group=cluster.nodes['control-plane'].include_group(
+                        cluster.nodes.get('worker')).get_ordered_members_list(
+                        provide_node_configs=True)
+            for node in node_group:
+                if "control-plane" in node["roles"]:
+                    control_plane = node
+                else:
+                    control_plane = cluster.nodes["control-plane"].get_first_member(provide_node_configs=True)
+                kubeadm_flags_file = "/var/lib/kubelet/kubeadm-flags.env"
+                pause_version = cluster.globals['compatibility_map']['software']['pause'][target_kubernetes_version]['version']
+                kubeadm_flags = node['connection'].sudo(f"cat {kubeadm_flags_file}",
+                                        is_async=False).get_simple_out()
+                updated_kubeadm_flags = re.sub(r'pause:.*\..*', f"pause:{pause_version}", kubeadm_flags)
```

Comment:
This removes trailing `"` from the kubeadm-flags.env. Instead of regex, I suggest to reuse `migrate_cri._config_changer` and `sandbox = sandbox[:param_begin_pos] + ":" + str(pause_version)` calculated earlier.

Reply:
Fixed the regex as ` '(pause:.*\.[^"]*)', [^"]* ` ensuring that the match ends before the trailing double quotes.
In this way, the trailing " will not be removed.

Reply:
This still removes all other properties after `--pod-infra-container-image` if they exist. Please do not use regex, and use already existing `migrate_cri._config_changer` instead. You may move that method to kubemarine/kubernetes/__init__.py.

Reply:
changes made to use config_changer() instead of regex and moved config_changer to kubemarine/kubernetes/__init__.py.